File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// Instantiate functions from below
int count_ltrs(char text[]);
int count_words(char text[]);
int count_sent(char text[]);

int main(void)
{
    // Set up the text array
    char text[500];
    do
    {
        // Prompt user for text
        printf("Text: ");
        fgets(text, sizeof(text), stdin);
    }
    while (strlen(text) - 1 == 0);

    // Use the functions to get counts
    int Ltrs = count_ltrs(text);
    int Wrds = count_words(text);
    int Sents = count_sent(text);

    // Print out the counts
    // printf("%i Letter(s)\n", Ltrs);
    // printf("%i Word(s)\n", Wrds);
    // printf("%i Sentence(s)\n", Sents);

    // Get L and S for Coleman-Liau Index formula
    float L = (float) Ltrs / Wrds * 100;
    float S = (float) Sents / Wrds * 100;

    // Apply the Coleman-Liau Index formula
    float index = (0.0588 * L) - (0.296 * S) - 15.8;

    // Get grade level
    int grade = round(index);

    // Display grade level
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}

// Functionn to get letter counts
int count_ltrs(char text[])
{
    // Get count of characters
    int num_char = 0;

    // Iterate through characters to get count
    for (int i = 0; i < strlen(text); i++)
    {
        // Condition to make sure it is a letter based on ASCII Decimal
        if ((text[i] > 64 && text[i] < 91) || (text[i] > 96 && text[i] < 123))
        {
            // Increase by one each time the above condition is True
            num_char++;
        }
    }
    return num_char;
}

// Function to get word counts
int count_words(char text[])
{
    // Get count of words
    int words = 1;

    // Iterate through characters to get count
    for (int i = 0; i < strlen(text); i++)
    {
        // Condition to get number of spaces based on ASCII Decimal
        if (text[i] == 32)
        {
            // Increase by one each time the above condition is True
            words++;
        }
    }
    return words;
}

// Function to get sentence counts
int count_sent(char text[])
{
    // Get counts of sentences
    int sent = 0;

    // Iterate through characters to get count
    for (int i = 0; i < strlen(text); i++)
    {
        // Condition to get just the . ! ? out based on ASCII Decimal
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            // Increase by one each time the above condition is True
            sent++;
        }
    }
    return sent;
}

File: pset6/readability/readability.py
# Imports
import re
from cs50 import get_string


# Create a function to count the letters in the text
def cnt_ltrs(text):
    # Create a counter for number of characters
    num_char = 0

    # Iterate through characters to get count of letters only
    for c in range(len(text)):
        if re.match(r'[A-Za-z]', text[c]):
            num_char += 1

    return num_char

# # Test my function for functionality
# test_char_cnt = cnt_ltrs('I love cats@!')
# print(test_char_cnt)


# Create a function to count the words in the text
def cnt_words(text):
    words = text.split(' ')
    return len(words)

# # Test my function for functionality    
# test_wrd_cnt = cnt_words('I love cats@!')
# print(test_wrd_cnt)


# Create a function to get sentance count
def sent_cnt(text):
    sents = list(map(str.strip, re.split(r'[.!?](?!$)', text)))
    return len(sents)

# # Test my function for functionality    
# test_sent_cnt = sent_cnt('I love cats@! Dogs are great too.')
# print(test_sent_cnt)


# Prompt user to enter some text
text = get_string('Text: ')

ltrs = cnt_ltrs(text)  # get letter count
wrds = cnt_words(text)  # get word count
sents = sent_cnt(text)  # get sentance count

# Get L and S for Coleman-Liau Index formula
L = ltrs / wrds * 100
S = (sents + 1) / wrds * 100

# Apply the Coleman-Liau Index formula
index = (0.0588 * L) - (0.296 * S) - 15.8

# Get the grade level rounded
grade = round(index)

if grade < 1:
    print('Before Grade 1')

elif grade >= 16:
    print('Grade 16+')

else:
    print(f'Grade {grade}')
